src/mutate.rs

```diff
@@ -102,8 +118,92 @@ impl Client {
             .await?
             .into_inner();
 
-        self.collection_cache.update_timestamp(&collection_name, result.timestamp);
+        self.collection_cache
+            .update_timestamp(&collection_name, result.timestamp);
+
+        Ok(result)
+    }
+
+    pub async fn delete(
+        &self,
+        collection_name: impl Into<String>,
+        options: &DeleteOptions,
+    ) -> Result<crate::proto::milvus::MutationResult> {
+        if options.ids.len() > 0 && options.filter.len() > 0 {
+            return Err(Error::InvalidParameter(
+                "len(filter)".to_owned(),
+                options.filter.len().to_string(),
+            ));
+        }
+
+        if options.ids.len() == 0 && options.filter.len() == 0 {
+            return Err(Error::InvalidParameter(
+                "len(filter)".to_owned(),
+                options.filter.len().to_string(),
+            ));
+        }
+
+        let collection_name = collection_name.into();
+
+        let mut expr = match options.filter.len() {
+            0 => {
+                let collection = self.collection_cache.get(&collection_name).await?;
+                let pk = collection.fields.iter().find(|f| f.is_primary_key).unwrap();
+
+                let mut expr = String::new();
+                expr.push_str(&pk.name);
+                expr.push_str(" in [");
+                match (pk.dtype, options.ids.clone()) {
+                    (DataType::Int64, ValueVec::Long(values)) => {
+                        for (i, v) in values.iter().enumerate() {
+                            if i > 0 {
+                                expr.push_str(",");
+                            }
+                            expr.push_str(format!("{}", v).as_str());
+                        }
+                        expr
+                    }
+
+                    (DataType::VarChar, ValueVec::String(values)) => {
+                        for (i, v) in values.iter().enumerate() {
+                            if i > 0 {
+                                expr.push_str(",");
+                            }
+                            expr.push_str(v.as_str());
+                        }
+                        expr
+                    }
+
+                    _ => {
+                        return Err(Error::InvalidParameter(
+                            "pk type".to_owned(),
+                            pk.dtype.as_str_name().to_owned(),
+                        ));
+                    }
+                }
+            }
```

Comment:
prefer to organize these codes to a member function (eg. makeExpr) to make it clearer